client.go

```diff
@@ -268,6 +276,23 @@ func CompressionCodec(codec string) Option {
 	}
 }
 
+// ZooKeeperDialer will return an option to pass the given dialer function
+// into the ZooKeeper client Connect() call, which allows for customizing
+// network connections.
+func ZooKeeperDialer(dialer gzk.Dialer) Option {
```

Comment:
I'd rather not make a type from github.com/go-zookeeper/zk an explicit part of the API of gohbase. If github.com/go-zookeeper/zk changes the Dialer type or adds a ContextDialer for example then we would have to change gohbase as well. Or gohbase decides to use a different zookeeper library in the future then this API will have to change.

What I would prefer is we make the region Dialer and the zookeeper Dialer the same type. And I think it makes more sense to make it a function rather than an interface (this is consistent with the zookeeper library and with gRPC).

The function should be: `func(ctx context.Context, network, addr string) (net.Conn, error)`. This works easy enough for the region dialer. For the zookeeper dialer you have to do a little work to convert the context dialer func to the zk.Dialer signature:

```go
func makeZKDialer(ctxDialer func(ctx context.Context, network, addr string) (net.Conn, error)) zk.Dialer {
    return func(network, addr string, timeout time.Duration) (net.Conn, error) {
        ctx, cancel := context.WithTimeout(context.Background(), timeout)
        defer cancel()
        return ctxDialer(ctx, network, addr)
    }
}
```

Reply:
will do.